Bio/SeqRecord.py

```diff
@@ -224,6 +224,13 @@ def __init__(
             raise TypeError("name argument should be a string")
         if not isinstance(description, str):
             raise TypeError("description argument should be a string")
+
+        # If seq is a string, other operations (such as saving from AlignIO)
+        # will fail. Either convert to Seq or raise an error
+        # see "Bio/SeqIO/Interfaces.py:_get_seq_string" for example.
+        if isinstance(seq, str):
+            seq = Seq(seq)
+
```

Comment:
I'd rather the ``SeqRecord`` raised a ``TypeError`` if the sequence argument was a string (or not an instance of Seq or MutableSeq).

Reply:
I am happy to make the change. I originally decided against that because the constructor has type "seq: Optional[Union["Seq", "MutableSeq", str]]", which suggests that is fine to use strings (meaning lots of people might be relying on this being allowed). Lots of code might break if we go that way. Just raising the concern (I don't mind it personally).

Reply:
Oh - good catch. That type annotation is new (not present in Biopython 1.81), and ought to be fixed too.

Reply:
Sounds good, I will update this PR accordingly (Y)